core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -63,12 +62,12 @@ private[zio] final class FiberContext[E, A](
   private[this] val traceEffects: Boolean =
     traceExec && platform.tracing.tracingConfig.traceEffectOpsInExecution
 
-  private[this] val fiberId         = FiberContext.fiberCounter.getAndIncrement()
   private[this] val stack           = Stack[Any => IO[Any, Any]]()
   private[this] val environments    = Stack[AnyRef](startEnv)
   private[this] val executors       = Stack[Executor](startExec)
   private[this] val interruptStatus = StackBool(startIStatus.toBoolean)
-  private[this] val supervised      = Stack[SuperviseStatus](startSStatus)
+  private[this] val children        = new AtomicReference(Set[FiberContext[_, _]]())
```

Comment:
Long term this could be replaced with a more efficient mutable Set, right?

Reply:
Yes, although when the children are retrieved, then you have to copy the whole thing into an immutable set, which probably implies making that retrieval a separate operation (now it is bundled into descriptor).

Reply:
Agreed on that. Retrieving the children would also be a far less frequent operation after this is merged.